### ISSUE TITLE
Audio won't reinitialize when driver initialization failed

### ISSUE THREAD

**Issue by Zapeth** [mpv-player/mpv#8579]
### Important Information

Provide following Information:
- mpv version: mpv-0.33.0-x86_64
- Platform and Version: Windows 7 x64
- Source of the mpv binary: sourceforge.net

### Reproduction steps

Switch to a non-available audio device during playback and try to switch back to an available one (using the method described in https://github.com/mpv-player/mpv/issues/2398)

### Expected behavior

Audio is restored when the command to switch to an available audio device is issued.

### Actual behavior

Audio stays muted once audio driver failed to initialize (looking at the log messages it seems like `ao_init_best` is not called anymore when trying to use a different device).

### Log file
```
[ao/wasapi] Failed to find device '<device-id>'
[ao] Failed to initialize audio driver 'wasapi'
[ao] This audio driver/device was forced with the --audio-device option.
[ao] Try unsetting it.
Could not open/initialize audio device -> no sound.
Audio: no audio
```

### Sample files

entry in input.conf: `a  cycle_values options/audio-device wasapi/<device-id> wasapi/<device-id2>`

**Comment by Dudemanguy** [comment]
>looking at the log messages it seems like ao_init_best is not called anymore when trying to use a different device

Yeah that's correct. No bug here; this is by design. When you write a value to a specific ao or audio-device during runtime like this, mpv will try to use it and there is no logic for recovering or anything like that. Autoprobing is only done during the initialization step and even then it's only done if such options are not explicitly set.

**Comment by bitingsock** [comment]
While the manual mentions not parsing the message log, it does provide for very useful catches sometimes. This will reset your device to auto if it sees the message:

```
mp.enable_messages("error")
mp.register_event("log-message", function(event)
	if event.text:find("Try unsetting it") then
		deviceList = mp.get_property_native("audio-device-list")
		mp.set_property("audio-device", "auto")
		mp.set_property("aid", aid)
	end
end)
```
